File: Part 2 - Regression/Section 6 - Polynomial Regression/regression_template.R
# Regression Template

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


# Fitting Regression Model to the dataset
# Create your Regression here


# Predicting a new result
predict(regressor, newdata = data.frame(Level = 6.5))

# Visualizing the Regression Model results (for higher resolution and smoother curve)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), 
             colour = 'red') + 
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))), 
            colour = 'blue') + 
  ggtitle('Truth or Bluff (Regression Model)') + 
  xlab('Level') +
  ylab('Salary')

File: Part 2 - Regression/Section 6 - Polynomial Regression/regression_template.py
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset =  pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values


# Splitting the dataset into the Training set and Test set
"""rom sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""



# Fitting the regression model to the dataset 
# Create your regressor here

# Predicting a new result
y_pred = regressor.predict(6.5)

# Visualiazing the Regression Result
plt.scatter(X, y, color='red')
plt.plot(X, lin_reg_2.predict(regressor.fit_transform(X)), color='blue')
plt.title('Truth or Bluff (Regression Model)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Predecting a new result in polynomial regression
lin_reg_2.predict(regressor.fit_transform([[6.5]]))










